shared/src/main/java/uk/gov/di/authentication/shared/services/DynamoAuthenticationAttemptsService.java

```diff
@@ -13,59 +13,74 @@ public DynamoAuthenticationAttemptsService(ConfigurationService configurationSer
     }
 
     public void addCode(
-            String attemptIdentifier, long ttlSeconds, String code, String authenticationMethod) {
+            String internalSubjectId,
+            long ttlSeconds,
+            String code,
+            String authenticationMethod,
+            String journeyType) {
         long ttlEpochSeconds =
                 NowHelper.nowPlus(ttlSeconds, ChronoUnit.SECONDS).toInstant().getEpochSecond();
 
         var authenticationAttempt =
-                get(attemptIdentifier)
+                get(internalSubjectId, authenticationMethod + "_" + journeyType)
                         .orElse(new AuthenticationAttempts())
-                        .withAttemptIdentifier(attemptIdentifier)
+                        .withInternalSubjectId(internalSubjectId)
                         .withCode(code)
                         .withAuthenticationMethod(authenticationMethod)
-                        .withTimeToLive(ttlEpochSeconds);
+                        .withTimeToLive(ttlEpochSeconds)
+                        .withJourneyType(journeyType);
 
         update(authenticationAttempt);
     }
 
     public void createOrIncrementCount(
-            String attemptIdentifier, long ttl, String authenticationMethod) {
-        Optional<AuthenticationAttempts> authenticationAttempt = get(attemptIdentifier);
+            String internalSubjectId, long ttl, String authenticationMethod, String journeyType) {
+        Optional<AuthenticationAttempts> authenticationAttempt =
+                get(internalSubjectId, authenticationMethod + "_" + journeyType);
```

Comment:
I think it's worth introducing a helper function somewhere to replace all the cases we manually do this concatenation (here and in the authentication attempts method) so that it's only defined in one place